app/javascript/packs/play_book.jsx

```diff
@@ -133,15 +123,17 @@ export default class PlayBook extends Component {
     this.setState({ isFull: true });
   }
 
-  handleStopDrawing() {
-    this.setState(previousState => ({ isDrawing: !previousState.isDrawing }));
-  }
-
   handleStartDrowingArrows(dashed) {
-    this.setState(previousState => ({
-      isDrawingArrows: !previousState.isDrawingArrows,
-      dashed: dashed
-    }));
+    if (this.state.dashed === dashed) {
+      this.setState(previousState => ({
+        isDrawingArrows: !previousState.isDrawingArrows,
+      }))
+    } else {
+      this.setState({
+        isDrawingArrows: true,
+        dashed: dashed
+      })
+    }
   }
 
   handleUndo() {
```

Comment:
this function is growing and it will be very hard to maintain it. Consider splitting it into smaller functions and try to use names for actions instead of meaningless numbers